block-producer/src/block_builder/prover/mod.rs

```diff
@@ -57,6 +71,51 @@ begin
     drop
     # => [ROOT_{n-1}]
 end
+
+#! Compute the note root
+#! 
+#! Stack: [num_notes_updated, SMT_EMPTY_ROOT, note_key_0, NOTE_HASH_0, ... , note_key_{n-1}, NOTE_HASH_{n-1}]
+#! Output: [NOTES_ROOT]
+proc.compute_note_root
```

Comment:
I'd probably add a bit more explanation here because it is not immediately clear that the way we compute notes root is by inserting roots of batch note roots into an SMT at depth 8.